ironic/drivers/modules/drac/management.py

```diff
@@ -308,7 +331,143 @@ class DracRedfishManagement(redfish_management.RedfishManagement):
     specific incompatibilities and introduction of vendor value added
     should be implemented by this class.
     """
-    pass
+    @METRICS.timer('DracRedfishManagement.clear_job_queue')
+    @base.clean_step(priority=0)
+    def clear_job_queue(self, task):
+        """Clear the job queue.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        self.get_redfish_info(task)
+        jobs_url = 'https://%s%s' % (self.redfish_ip, _JOBS_BASE_URL)
+        job_response = requests.get(
+            jobs_url,
+            auth=(self.redfish_username, self.redfish_password), verify=False)
+
+        jobs_data = job_response.json()
+        job_list = re.findall("JID_.+?'", str(jobs_data))
+        if job_list is None:
+            LOG.info(
+                'Job queue already cleared for iDRAC %s'
+                % (self.redfish_ip))
+            return None
+
+        try:
+            for job_id in job_list:
+                job_id = job_id.strip("'")
+                job_url = '%s/%s' % (jobs_url, job_id)
+                delete_response = requests.delete(
+                    job_url, headers=_HEADERS, verify=False,
+                    auth=(self.redfish_username, self.redfish_password))
+                LOG.info(
+                    'Job queue for iDRAC %s successfully cleared' %
+                    (self.redfish_ip))
+        except drac_exceptions.BaseClientException as exc:
+            LOG.error('DRAC driver failed to clear the job queue for node '
+                      '%(node_uuid)s. Reason: %(error)s.',
+                      {'node_uuid': task.node.uuid, 'error': exc})
+            raise exception.DracOperationError(error=exc)
+
+    @METRICS.timer('DracRedfishManagement.reset_idrac')
+    @base.clean_step(priority=0)
+    def reset_idrac(self, task):
+        """Reset the iDRAC.
+
+        :param task: a TaskManager instance containing the node to act on.
+        :returns: None if it is completed.
+        :raises: DracOperationError on an error from python-dracclient.
+        """
+        self.get_redfish_info(task)
+        reset_idrac_url = 'https://%s%s' % (self.redfish_ip, _RESET_URL)
+        payload = {"ResetType": "GracefulRestart"}
+
+        try:
+            reset_response = requests.post(
+                reset_idrac_url,
+                data=json.dumps(payload), headers=_HEADERS, verify=False,
+                auth=(self.redfish_username, self.redfish_password))
+
```

Comment:
We need to check the http error code returned and raise an exception on failure.

Reply:
Addressed, moved low level redfish code to sushy-oem-idrac repo and added check for returned response code value.